letsencrypt/client.py

```diff
@@ -237,12 +240,20 @@ def obtain_certificate(self, domains):
         :rtype: tuple
 
         """
+        authzr = self.auth_handler.get_authorizations(
+                domains,
+                self.config.allow_subset_of_names)
+
+        domains = [a.body.identifier.value.encode('ascii', 'ignore')
```

Comment:
encode('ascii') itself didn't fail, but I had to encode the domains since they were originally encoded in unicode. Boulder integration test failed as a result.